Not retrying the command

**Describe the bug**
HI,
here is my test snippet 
```yaml
- name: Build with Maven
  uses: nick-invision/retry@v2.4.0
  with:
    timeout_seconds: 60
    max_attempts: 3
    retry_wait_seconds: 20
    retry_on: timeout
    command: mvn -Dmaven.wealthcentral.full.build=true clean install site
```

I tried timeout_seconds=60 timeout_minutes=1, still not working . I tried retry_on: error and it restarted the command, but I only wants to restart the command on timeout. Am I doing something wrong, or misunderstand the retry_on: timeout function ?

**Expected behavior**
I would expect the command to restart "mvn -Dmaven.wealthcentral.full.build=true clean install site". but I only get the error and the step fails.

**Screenshots**
![image](https://user-images.githubusercontent.com/82996400/126731763-33a6d799-697d-483f-9075-2e28ee004e85.png)

**Logs**
https://pipelines.actions.githubusercontent.com/VZK7aeekU2uaNe4Ic693P6kodHwzQMznZWFFtaEw4BXgNmLWia/_apis/pipelines/1/runs/4111/signedlogcontent/3?urlExpires=2021-07-23T02%3A43%3A50.6523982Z&urlSigningMethod=HMACV1&urlSignature=q0FhuLAf1914%2FeQ%2FEOZJHeYaabiI5OlstxPwr3RiUvY%3D

This sounds similar to what I am seeing...

I am using as follows:

```
      - name: Run Tests - ${{ matrix.suite }}
        uses: nick-fields/retry@v2
        timeout-minutes: 33 # 3 attempts at 10+1 mins max each
        env:
          DATABASE_IMAGE_NAME: ${{ matrix.databaseImageName }}
        with:
          # This step should only take ~5 mins at most to complete, but sometimes locks up
          timeout_minutes: 10
          max_attempts: 3
          retry_on: timeout
          command: make test SUITE=${{ matrix.suite }}
```

And there is the log output, with only a single timeout indicated, and no retires occouring

```
Thu, 24 Nov 2022 17:21:43 GMT Run nick-fields/retry@v[2](https://github.com/wmde/wikibase-release-pipeline/actions/runs/3488327445/jobs/5948299853#step:4:2)
Thu, 24 Nov 2022 17:21:43 GMT   with:
Thu, 24 Nov 2022 17:21:4[3](https://github.com/wmde/wikibase-release-pipeline/actions/runs/3488327445/jobs/5948299853#step:4:3) GMT
    timeout_minutes: 10
Thu, 24 Nov 2022 17:21:43 GMT     max_attempts: 3
Thu, 2[4](https://github.com/wmde/wikibase-release-pipeline/actions/runs/3488327445/jobs/5948299853#step:4:4) Nov 2022 17:21:43 GMT
    retry_on: timeout
Thu, 24 Nov 2022 17:21:43 GMT     command: make test SUITE=repo
Thu, 24 Nov 2022 17:21:43 GMT     retry_wait_seconds: 10
Thu, 24 Nov 2022 17:21:43 GMT     polling_interval_seconds: 1
Thu, 24 Nov 2022 1[7](https://github.com/wmde/wikibase-release-pipeline/actions/runs/3488327445/jobs/5948299853#step:4:7):21:43 GMT
    warning_on_retry: true
Thu, 24 Nov 2022 17:21:43 GMT     continue_on_error: false
Thu, 24 Nov 2022 17:21:43 GMT   env:
Thu, 24 Nov 2022 17:21:43 GMT     env_file: versions/wmde[9](https://github.com/wmde/wikibase-release-pipeline/actions/runs/3488327445/jobs/5948299853#step:4:9).env
Thu, 24 Nov 2022 17:21:43 GMT     DATABASE_IMAGE_NAME: mariadb:[10](https://github.com/wmde/wikibase-release-pipeline/actions/runs/3488327445/jobs/5948299853#step:4:10).9
Thu, 24 Nov 2022 17:21:43 GMT cd test && bash test_stop.sh ""
Thu, 24 Nov 2022 17:21:44 GMT No containers to terminate.
Thu, 24 Nov 2022 17:21:44 GMT bash test/test_suite.sh repo
Thu, 24 Nov 2022 [17](https://github.com/wmde/wikibase-release-pipeline/actions/runs/3488327445/jobs/5948299853#step:4:18):21:44 GMT
#########################################
Thu, 24 Nov 2022 17:21:44 GMT #########################################
Thu, 24 Nov 2022 17:21:44 GMT ####         Start testing! 🤞       ####

...snip...

Thu, 24 Nov 2022 17:25:06 GMT [0-1] RUNNING in chrome - /specs/repo/property.js
Thu, 24 Nov 2022 17:31:44 GMT make: *** wait: No child processes.  Stop.
Thu, 24 Nov 2022 17:31:44 GMT make: *** Waiting for unfinished jobs....
Thu, 24 Nov 2022 17:31:44 GMT make: *** wait: No child processes.  Stop.
Thu, 24 Nov 2022 17:31:44 GMT 1
Thu, 24 Nov 2022 17:31:54 GMT Error: Timeout of 600000ms hit
```

This is from https://github.com/wmde/wikibase-release-pipeline/actions/runs/3488327445/jobs/5948299853

--- a/src/index.ts
+++ b/src/index.ts
@@ -71,6 +71,7 @@ async function runCmd(attempt: number, inputs: Inputs) {
 
   exit = 0;
   done = false;
+  let timeout = false;
 
   debug(`Running command ${inputs.command} on ${OS} using shell ${executable}`);
   const child =
@@ -94,6 +95,11 @@ async function runCmd(attempt: number, inputs: Inputs) {
       return;
     }
 
+    // On Windows signal is null.
+    if (timeout) {
+      return;
+    }
+
     if (code && code > 0) {
       exit = code;
     }
@@ -106,6 +112,7 @@ async function runCmd(attempt: number, inputs: Inputs) {
   } while (Date.now() < end_time && !done);
 
   if (!done && child.pid) {
+    timeout = true;
     kill(child.pid);
     await retryWait(ms.seconds(inputs.retry_wait_seconds));
     throw new Error(`Timeout of ${getTimeout(inputs)}ms hit`);

--- a/src/index.ts
+++ b/src/index.ts
@@ -88,13 +88,16 @@ async function runCmd(attempt: number, inputs: Inputs) {
   child.on('exit', (code, signal) => {
     debug(`Code: ${code}`);
     debug(`Signal: ${signal}`);
-    if (code && code > 0) {
-      exit = code;
-    }
+
     // timeouts are killed manually
     if (signal === 'SIGTERM') {
       return;
     }
+
+    if (code && code > 0) {
+      exit = code;
+    }
+
     done = true;
   });
 

